src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -73,20 +79,72 @@ public ImmutableArray<DiagnosticData> GetBuildErrors()
         public bool IsSupportedDiagnosticId(ProjectId projectId, string id)
             => BuildInprogressState?.IsSupportedDiagnosticId(projectId, id) ?? false;
 
+        private void OnBuildProgressChanged(InProgressState? state, BuildProgress buildProgress)
+        {
+            if (state != null)
+            {
+                _lastBuiltResult = state.GetBuildErrors();
+            }
```

Comment:
so if state is null, it's intentional that we don't overwrite this (As opposed to writing an empty collection to it)?

Reply:
I really don't know when the state can be null, so I am just retaining the existing semantics of this code. When attempting to debug this code path, state was always non-null, so I am presuming the null possibility is either existing defensive code path or some corner case.

Reply:
btw, this is worth documenting (even if you don't know the answer and are jsut preserving things).  that helps future maintenance when the next person has to ask "is this intentional or not?"